services/server/getEvents.ts

```diff
@@ -0,0 +1,116 @@
+import client from "@/tina/client";
+
+export const getPastEvents = async (top, presenterName) => {
+  console.log("presenterName", presenterName);
+  const eventClient = await client.queries.getPastEventsQuery(
+    formatEventParams(top, presenterName)
+  );
+  return await fetchEventsWithClient(eventClient, presenterName, top);
+};
+
+export const getUpcomingEvents = async (top, presenterName) => {
+  const eventClient = await client.queries.getFutureEventsQuery(
+    formatEventParams(top, presenterName)
+  );
+  return await fetchEventsWithClient(eventClient, presenterName, top);
+};
+
+const formatEvent = (event) => {
+  const url = event.url && fixRelativeUrl(event.url);
+  return {
+    Presenter: event.presenterName,
+    Url: {
+      Description: event.title, // this is alt text for the link, ask matt what he wants to do for it
+      Url: url,
+    },
+    title: event.title,
+    Thumbnail: {
+      Url: fixRelativeUrl(event.thumbnail),
+      Description: event.thumbnailDescription,
+    },
+    PresenterProfileUrl:
+      event.presenterProfileUrl && fixRelativeUrl(event.presenterProfileUrl),
+    StartDateTime: event.startDateTime,
+    EndDateTime: event.endDateTime,
+    StartShowBannerDateTime: event.startShowBannerDateTime,
+    EndShowBannerDateTime: event.endShowBannerDateTime,
+    LiveStreamUrl: event.youTubeId
+      ? `https://www.youtube.com/watch?v=${event.youTubeId}`
+      : null,
+    CalendarType: event.calendarType,
+    City: event.city,
+    Enabled: event.enabled,
+    EventShortDescription: descriptionToHTML(event.description),
+    ShowOnPowerSessions: "No",
+    NoteInternal: event.internalNote,
+    HasVideo: event.youTubeId ? "Yes" : "No",
+    YouTubeId: event.youTubeId,
+    PresenterDescription:
+      event.presenterList && event.presenterListlength > 0
+        ? event.presenterList[0].presenter.about
+        : null,
+  };
+};
+export const fetchEventsWithClient = async (
+  eventClient,
+  presenterName: string | undefined,
+  top
+) => {
+  if (top) {
+    top = parseInt(top);
+  }
+  const events = [];
+  // TODO: remove client side after fixing events with multiple presenters in the name
+  for (const event of eventClient.data.eventsCalendarConnection.edges) {
+    if (
+      !presenterName ||
```

Comment:
the logic inside your "if" should be extracted in its own method - this if statement is too complex.